platoon/channel.py

```diff
@@ -248,20 +259,34 @@ def init_control_sock(self, port):
         self.cpoller = zmq.Poller()
         self.cpoller.register(self.csocket, zmq.POLLIN)
 
-    def init_shared_params(self, job_name, params, param_sync_rule,
-                           cleanup=False):
+    def _mmap(self, length=0, prot=0x3, flags=0x1, fd=0, offset=0):
+        _ffi = cffi.FFI()
+        _ffi.cdef("void *mmap(void *, size_t, int, int, int, size_t);")
+        _lib = _ffi.dlopen(None)
+
+        addr = _ffi.NULL
+
+        m = _lib.mmap(addr, length, prot, flags, fd, offset)
+        if m == _ffi.cast('void *', -1):
+            raise OSError(_ffi.errno, "for mmap")
+        return _ffi.buffer(m, length)
```

Comment:
Defining a mapping with cffi is not free so this is why I do it at module level.  Now you re-do it every this this is called.

It may not be so bad since it won't be called often, but why did you change that?